### PR TITLE
Section banner

### PR DESCRIPTION
**Pull request by gcater**
When the user is on the sectionId/index page rather than have a menu dropdown button this will be a bookDropdown button.

This will show all the chapters and sections of the book with links to them. 

https://www.loom.com/share/680898e7a3cd4f6ea9ec8e219bbc028f?sid=685b02d0-0c73-4fa6-af0e-b2c78ed6c29d

### CODE REVIEW

**Comment by Evanc123** [comment]
The dropdown is not pixel perfect

**Comment by gcater** [comment]
Screenshot 2024-04-18 at 2 04 35 PM

Thoughts? The font that The Network State uses is some paid one, so i'm just using font-serif.

**Comment by Evanc123** [comment]
Better -- looks like there is more vertical padding between the elements in the original tho

**Comment by gcater** [comment]
increased vertical and horizontal padding. 
Screenshot 2024-04-18 at 2 26 38 PM
Screenshot 2024-04-18 at 2 26 49 PM

**Comment by gcater** [comment]
Added some padding. Looks like the difference in font adds a slight offset

**Review by Evanc123 — APPROVED** [review]

File: src/components/DropDownBookButton.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                          key={sectionIndex}
+                          className="w-full border-t border-gray-300"
+                        >
+                          <div className="hover:bg-gray-50">
+                            <Link
+                              href={`/book/${bookId}/${chapter.id}/${section.id}`}
+                              className="cursor-pointer text-black"
+                            >
+                              <div className="py-2.5">
+                                <div className="px-5 font-serif">{section.title}</div>
```
Comment (Evanc123) [inline src/components/DropDownBookButton.tsx:83]: you just need one font-serif at the grandparent of all of these divs

File: src/components/DropDownBookButton.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const { data: book } = api.book.getBook.useQuery(bookId);
+  if (!book) return <div>No book found</div>;
+  const toggleDropdown = () => setIsDropdownOpen(!isDropdownOpen);
+  return (
+    <div className="absolute right-4 top-[10px]">
+      <div
+        className="svg-container cursor-pointer" 
+        onClick={toggleDropdown}
+        style={{ display: "inline-block", lineHeight: 0 }}
```
Comment (Evanc123) [inline src/components/DropDownBookButton.tsx:14]: can this not be done with tailwind?

File: src/pages/book/[bookid]/index.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default function Home() {
         <TopBanner bookId={bookid} />
         <div className="flex min-h-screen flex-col items-center justify-center">
           <div className="flex w-full">
-            <div className="w-1/2">{BookUpdate({ bookId: bookid })}</div>
+            <div className="w-1/2">{BookUpdate({ bookId: bookid })}</div>
```
Comment (Evanc123) [inline src/pages/book/[bookid]/index.tsx:25]: these should be components, not functions, but we can fix that later

**Pull request merged**